storybook/qmlTests/TestComponents/tst_StatusChatInput.qml

```diff
@@ -1,21 +1,571 @@
 import QtQuick 2.14
+import QtQuick.Controls 2.14
 import QtQml 2.14
 import QtTest 1.0
 
 import utils 1.0
 import shared.status 1.0
 import shared.stores 1.0
 
+import TextUtils 1.0
+
 Item {
     id: root
     width: 600
     height: 400
 
+    Component {
+        id: componentUnderTest
+        StatusChatInput {
+            property var globalUtils: globalUtilsMock
+
+            width: parent.width
+            height: implicitHeight
+            anchors.bottom: parent.bottom
+            usersStore: QtObject {
+                property var usersModel: ListModel {}
+            }
+        }
+    }
+
+    TestCase {
+        name: "StatusChatInputInitialization"
+        when: windowShown
+
+        property StatusChatInput controlUnderTest: null
+
+        function init() {
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+        }
+
+        //Scenario: StatusChatInput initialisation
+        //Given a new instance of StatusChatInput
+        //When there is no keyboard input
+        //Then the StatucChatInput raw text is empty
+        //But the StatusChatInput is configured to support Rich Text
+        function test_empty_chat_input() {
+            verify(controlUnderTest.textInput.length == 0, `Expected 0 text length, received: ${controlUnderTest.textInput.length}`)
+            verify(controlUnderTest.getPlainText() == "", `Expected empty string, received: ${controlUnderTest.getPlainText()}`)
+            verify(controlUnderTest.textInput.textFormat == TextEdit.RichText, "Expected text input format to be Rich")
+            verify(controlUnderTest.textInput.text.startsWith("<!DOCTYPE HTML PUBLIC"), "Expected text input format to be Rich")
+        }
+    }
+
+    TestCase {
+        id: statusChatInputKeyboardInputExpectedAsText
+        name: "StatusChatInputKeyboardInputExpectedAsText"
+        when: windowShown
+
+        property StatusChatInput controlUnderTest: null
+
+        function init() {
+            Utils.globalUtilsInst = globalUtilsMock
+            controlUnderTest = createTemporaryObject(componentUnderTest, root)
+        }
+
+        //Scenario: StatusChatInput will display any typed Ascii visible character
+        //Given a new instance of StatusChatInput
+        //When the user is typing any Ascii visible character as <text>
+        //Then the text is displayed as it is typed
+        //And the input text is not modified by mentions processor
+        //And the input text is not modified by emoji processor
+        //
+        //Example:
+        //text
+        //(!"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\]^_`abcdefghijklmnopqrstuvwxyz{|}~)
+        function test_keyboard_any_ascii_input() {
```

Comment:
I'd say extend this to UTF-8 characters as well... not all obviously, but we can try a few randomly selected ones, wdyt?

Reply:
keyClick has an assert if we try to input non ascii and this is why I was limited to this set.
We could test a larger set when we test the parser that is meant to parse the message received from backend (Ex: on message edit)